libs/server-sdk/src/data_systems/background_sync/sources/README.md

```diff
@@ -0,0 +1,22 @@
+## Data Source Implementations
+
+This directory contains implementations of `IDataSource`.
+
+There are two primary schemes supported by the server-side SDK.
+
+One is to receive push-updates from a web-service (via Server-Sent Events). The other is to pull
+data from a database as needed.
+
+### Push Data Source
+
+The Push data source is able to operate in "streaming" or "polling" mode. In either mode, flag
+updates are delivered in the background from the perspective of the application.
+
+### Pull Data Source
+
+The Pull data source fetches flag data on-demand from a database. Because this is likely slow,
+it operates a read-through cache wherein each stored item has a TTL indicating when it should be
+re-fetched.
```

Comment:
The terminology / distinctions here seem a little confusing.

Is the polling data model considered a "push data source"? That doesn't seem right. But the description of pull doesn't sound like you intended it to include polling.

Reply:
Yea, discussed this with Ryan and I see how it's confusing. I'm trying to draw a distinction between the interface implemented by persistent stores, and that of our streaming/polling sources.

 
The terminology comes from observing the data flow from the source. With polling/streaming mode, the data is pushed into the memory cache - it's asynchronous. With the persistent stores, it's pulled into the memory cache on demand - synchronously. 

 I was looking for terms that:
- Are symmetric 
- Short
- Have a commonly known meaning
- Don't mention streaming/polling/persistent stores directly